frontend/packages/kubevirt-plugin/src/components/modals/boot-order-modal/boot-order-modal.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Modal, Button } from '@patternfly/react-core';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { VMLikeEntityKind, BootableDeviceType } from '../../../types';
+import { getVMLikeModel, getDevices } from '../../../selectors/vm';
+import { PatchBuilder, PatchOperation } from '../../../k8s/utils/patch';
+import { getVMLikePatches } from '../../../k8s/patches/vm-template';
+import { BootOrder } from '../../boot-order';
+import { DeviceType } from '../../../constants';
+
+const modalTitle = 'Virtual machine boot order';
+
+export const BootOrderModal = ({
+  vmLikeEntity,
+  isOpen,
+  setOpen,
+  title = modalTitle,
+}: BootOrderModalProps) => {
+  const [devices, setDevices] = React.useState<BootableDeviceType[]>(getDevices(vmLikeEntity));
+   
+  // Update device list on vmLikeEntity change.
+  React.useEffect(() => {
+    const updatedDevices = getDevices(vmLikeEntity);
+
+    setDevices(updatedDevices);
```

Comment:
please look again at https://github.com/openshift/console/pull/3436#discussion_r348083655

we should ignore the changes of the boot order from somebody else. The newest bootorder should come from the dialog. I think user is not interested in boot order changes if s/he wishes it to change.

Reply:
ok ... will change to logic here ...

Reply:
update:
a - inform users if boot order changed outside.
b - on submit, only change the boot order, and only for currently available devices.